docs/Makefile

```diff
@@ -8,10 +8,54 @@ SPHINXBUILD   ?= sphinx-build
 SOURCEDIR     = .
 BUILDDIR      = _build
 
+JUPYTEXT      ?= jupytext
+NBCONVERT     ?= jupyter nbconvert
+DEMOFOLDER   = demos/demos
+OUTFOLDER    = demos/out
+
+# Script to add download buttons to the bottom of the rst files
+ADD_DOWNLOAD_BUTTONS ="$\
+\$$a$\
+:download:\`Download Python source code \<$$(basename $$1 .rst).py\>\`\n\n$\
+:download:\`Download as Jupyter Notebook \<$$(basename $$1 .rst).ipynb\>\`\n"
+
+# Script to add file names as labels to the top of the rst files
+ADD_FILE_NAME_LABEL = "1s/^/.. _$$(basename $$1 .rst):\n\n/"
+
 # Put it first so that "make" without argument is like "make help".
 help:
 	@$(SPHINXBUILD) -M help "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
 
+.PHONY: demos
+demos:
+	# 0. create the output folder and copy all Python source files over to it
+	mkdir -p $(OUTFOLDER)
+	cp -a $(DEMOFOLDER)/. $(OUTFOLDER)
+
+	# 1. find all Python demos and convert them to Jupyter notebooks
+	find $(DEMOFOLDER) -name '*.py' -exec sh -c '$(JUPYTEXT) --to ipynb $$1 --output "$(OUTFOLDER)/$$(basename $$1 .py).ipynb"' sh {} \;
```

Comment:
You can use shell's built-in [parameter expansion](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02) to avoid calling (external) `baseline`.